tests/test_utils_google_directions.py

```diff
@@ -327,28 +431,30 @@ def test_read_saved_api_results():
                                     'parsed_response': {'google_speed': 6.8, 'google_polyline': 'mvmyHpqYb@pA'}}})
 
 
-def test_queries_build_correctly_without_traffic():
+def test_queries_build_correctly_without_traffic_model_specified():
     request = google_directions.make_request(
         origin_attributes={'lat': 1, 'lon': 2},
         destination_attributes={'lat': 3, 'lon': 4},
         key='super_awesome_key',
-        traffic=False
+        departure_time=-1,
```

Comment:
**!!! SPOILER ALERT FOR UPCOMING UNIT TESTING SESSION !!!**

This test is doing a little bit too much, on the sly. The `None` traffic model param may well have been handled correctly, which I think is the intent of this test, but if the logic that turns a departure time of`-1` into `now` in the URL query string is broken, this test will still fail. As will a few other tests that are intended to test other things.

This is because we are implicitly asserting (at least) two different things at the same time:

- the traffic model param is missing from the URL query string (which we expect because we passed it as `None`)
- the departure time in the URL query string has been set to `now` (which we expect because we passed it as `-1`)

This test can therefore fail for two different reasons, only one of which is suggested by the test name `test_queries_build_correctly_without_traffic_model_specified`.

Add a separate test that asserts only that a departure time of `-1` becomes `now` in the query string. You may need to split the url/query string or else use more focussed assertions on the URL string, e.g. `assert "departure_time=now" in result.url`.

The same comment applies to all of the tests that are asserting against the full URL - to be really focused on a specific piece of logic, these tests should make assertions only about the corresponding query string param. The result should be that if, say, the code that turns a departure time of `-1` into a query string param of `now` is broken, only one test fails. I'd rather have multiple tests fail than none at all, of course, but the ideal situation is that one test fails and the name of the test alone makes it obvious what is broken.